pkg/api/helpers.go

```diff
@@ -427,3 +437,64 @@ func GetAffinityFromPodAnnotations(annotations map[string]string) (Affinity, err
 	}
 	return affinity, nil
 }
+
+// GetTolerationsFromPodAnnotations gets the json serialized tolerations data from Pod.Annotations
+// and converts it to the []Toleration type in api.
+func GetTolerationsFromPodAnnotations(annotations map[string]string) ([]Toleration, error) {
+	var tolerations []Toleration
+	if len(annotations) > 0 && annotations[TolerationsAnnotationKey] != "" {
+		err := json.Unmarshal([]byte(annotations[TolerationsAnnotationKey]), &tolerations)
+		if err != nil {
+			return tolerations, err
+		}
+	}
+	return tolerations, nil
+}
+
+// GetTaintsFromNodeAnnotations gets the json serialized taints data from Pod.Annotations
+// and converts it to the []Taint type in api.
+func GetTaintsFromNodeAnnotations(annotations map[string]string) ([]Taint, error) {
+	var taints []Taint
+	if len(annotations) > 0 && annotations[TaintsAnnotationKey] != "" {
+		err := json.Unmarshal([]byte(annotations[TaintsAnnotationKey]), &taints)
+		if err != nil {
+			return taints, err
+		}
+	}
+	return taints, nil
+}
+
+// TolerationToleratesTaint checks if the toleration tolerates the taint.
+func TolerationToleratesTaint(toleration Toleration, taint Taint) bool {
+	if len(toleration.Effect) != 0 && toleration.Effect != taint.Effect {
+		return false
+	}
+
+	//	return toleration.Key == taint.Key &&
+	//		(toleration.Operator == TolerationOpExists ||
+	//			((len(toleration.Effect) == 0 || toleration.Operator == TolerationOpEqual) &&
+	//				taint.Value == toleration.Value))
+	if toleration.Key != taint.Key {
+		return false
+	}
+	if (len(toleration.Operator) == 0 || toleration.Operator == TolerationOpEqual) && toleration.Value == taint.Value {
```

Comment:
The "right" way to implement "Defaults to Equal" here is though a defaulting function, see for example
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/defaults.go
(and all the other files that end in defaults.go)

However, I'm not sure defaulting really works for annotations. So I would just suggest to add a TODO here that says
```
// TODO: Use proper defaulting when Toleration becomes a field of PodSpec
```
(I was thinking of suggesting to add the TODO  in the comment for the Toleration.Operator field in types.go, but that gets turned into swagger documentation, and it wouldn't make any sense there.)